db/trie/mptrie/node.go

```diff
@@ -17,20 +17,28 @@ var ErrNoData = errors.New("no data in hash node")
 type (
 	keyType []byte
 
+	client interface {
```

Comment:
seems to me `merklePatriciaTrie` implements this interface, is there more purpose/use of this interface?

if that's the case, name it like `trieOps`? honestly `client` is a bit confusing in this context